pkg/operator/controller/canary/controller.go

```diff
@@ -44,6 +45,12 @@ const (
 	// canaryCheckFailureCount is how many successive failing canary checks should
 	// be observed before the default ingress controller goes degraded.
 	canaryCheckFailureCount = 5
+
```

Comment:
If this is a user-facing annotation, should this annoation be documented elsewhere?   Also, please add that the user must set the value as true or false, i.e. the annotation implies false if its value is not set to true.

Reply:
I enhanced the go-doc per your request. 
We don't expect users to want to opt-in to this functionality given the performance impact on the default cluster routes, so I think the documentation here is sufficient. This annotation would be useful for debugging a customer case via bugzilla or our support engineers, or for internal debugging / testing.